File: app/Sources/LocalCaptionKit/SessionFiles.swift
import Foundation

/// Filesystem operations on a session's saved transcript artifacts.
public enum SessionFiles {
    /// Delete a transcript `.txt` and its sibling `.json` sidecar together (SPEC-06 delete).
    /// Returns the URLs that were actually removed.
    @discardableResult
    public static func deleteTranscript(atTxtPath path: String) -> [URL] {
        let txt = URL(fileURLWithPath: path)
        let json = txt.deletingPathExtension().appendingPathExtension("json")
        let fm = FileManager.default
        var removed: [URL] = []
        for url in [txt, json] where fm.fileExists(atPath: url.path) {
            if (try? fm.removeItem(at: url)) != nil { removed.append(url) }
        }
        return removed
    }
}

File: windows/src/LocalCaption.Core/Data/SessionFiles.cs
namespace LocalCaption.Core.Data;

/// <summary>Filesystem operations on a session's saved transcript artifacts.</summary>
public static class SessionFiles
{
    /// <summary>
    /// Delete a transcript <c>.txt</c> and its sibling <c>.json</c> sidecar together
    /// (SPEC-06 delete). Returns the paths actually removed.
    /// </summary>
    public static IReadOnlyList<string> DeleteTranscript(string txtPath)
    {
        var jsonPath = Path.ChangeExtension(txtPath, ".json");
        var removed = new List<string>();
        foreach (var path in new[] { txtPath, jsonPath })
        {
            if (!File.Exists(path)) continue;
            try { File.Delete(path); removed.Add(path); } catch (IOException) { } catch (UnauthorizedAccessException) { }
        }
        return removed;
    }
}
